modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/scala/edu/gemini/seqexec/web/client/model/ModelOps.scala

```diff
@@ -81,13 +81,13 @@ object ModelOps {
     def nextStepToRun: Option[Int] =
       s.steps match {
         case x if x.forall(s => s.status === StepState.Pending && !s.skip) => Some(0) // No steps have been executed, start at 0
-        case x if x.forall(_.status === StepState.Completed)               => None // All steps have been executed
+        case x if x.forall(_.isFinished)                                   => None // All steps have been executed
         case x if x.exists(_.hasError)                                     => Option(x.indexWhere((s: Step) => s.hasError)).filter(_ =/= -1)
         case x if x.exists(s => s.status === StepState.Paused && !s.skip)  => Option(x.indexWhere((s: Step) => s.status === StepState.Paused)).filter(_ =/= -1)
-        case x                                                             => Option(x.indexWhere((s: Step) => s.status =/= StepState.Completed && !s.skip)).filter(_ =/= -1)
```

Comment:
This worked most of the time, because almost always if the `Step` was `Skipped`, then `skip` was `true`. The exception was if the Step was loaded as `Skipped` from the ODB.